src/components/RigisterModal/RigisterModal.jsx

```diff
@@ -0,0 +1,93 @@
+import { useEffect } from "react";
+import { useForm } from "../../hooks/useForm";
+import ModalWithForm from "../ModalWithForm/ModalWithForm";
+
+function RigisterModal({
+  isOpen,
+  handleModalClose,
+  handleModalToggle,
+  handleRegister,
+}) {
+  const { values, handleValueChange, errors, isValid, resetForm } = useForm({});
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    handleRegister(values);
+  };
+
+  useEffect(() => {
+    resetForm();
+  }, [isOpen]);
+
+  return (
+    <ModalWithForm
+      title="Sign up"
+      buttonText="Sign up"
+      naviText="Sign in"
+      isOpen={isOpen}
+      handleModalClose={handleModalClose}
+      handleModalToggle={handleModalToggle}
+      isBtnDisabled={!isValid}
+      onSubmit={handleSubmit}
+    >
+      <>
```

Comment:
`<>` empty tags are not needed if you have a common tag that wraps the whole markup of the component